### DISCUSSION TITLE
Add checkbox form control

### DISCUSSION THREAD

**Discussion by phorcys420** [espanso/espanso#1840] [Feature requests and Ideas]
Espanso lacks of a checkbox control, which would be really useful in cases where you only need to modify some parts of a text template.
I think a checkbox control would be a great addition !

---

Currently, the cleanest workaround is to use dropdowns, which still isn't great for UI and UX.
![image](https://github.com/espanso/espanso/assets/57866459/5a6a9b16-6a24-4b22-be7d-cc91e9ca4955)

**Comment by Rowyfo** [comment]
I agree. My form is very messy! This is how I choose up to 5 items from a list of 19 (Tags) as well as choosing up to 5 Who from a list of 9 (which would grow if it wasn't so painful). And sometimes it leaves white space when I don't want to add anything in that section.

![image](https://github.com/espanso/espanso/assets/69438918/1841892d-11b2-4f8a-9533-d2cfc257bb61)

**Comment by smeech** [comment]
What would you like the checkboxes to return? `true`/`false`? Because I think Espanso only returns strings, it would have to be text rather than proper boolean.

**Reply by Rowyfo** [reply]
In my case, I want it to print strings. So basically I have a list of words and if it applies, then I check it and it's printed.

**Reply by smeech** [reply]
So I guess you'd have to script that, and handle whatever the check-box fields put out.

**Reply by Rowyfo** [reply]
It's workable at present - I just feel like it could be much more efficient. I'm happy, for now, to deal with manually removing errant spaces and the like.

Basically, I defined the tags and then repeat them all in each drop-down, in cases where I want to choose more than one. Currently at 29 tags with scope to add more. Looks like I had a typo in my first message where I said I had 19.

**Reply by phorcys420** [reply]
> What would you like the checkboxes to return? `true`/`false`? 

I think they should return whatever you tell them to.
Something like
```yaml
  - trigger: ":example"
    form: |
      [[checkbox]]
    form_fields:
      checkbox:
        type: choice
        true: "The checkbox was checked"
        false: "The checkbox wasn't checked"
```

This helps with only using espanso without scripts, since scripts add some latency, yet you can still use scripts.

> I think Espanso only returns strings, it would have to be text rather than proper boolean.

It doesn't matter since they're passed as environment variables so they'll be strings anyways.
